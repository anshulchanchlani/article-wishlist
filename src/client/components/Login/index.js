import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser } from '../../actions/authActions'

import { withRouter, Redirect } from 'react-router-dom'
import './index.scss'
class Login extends Component {
    constructor(props) {
        super(props);
        this.state = { username: '', password: '' };
        this.onChange = this.onChange.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
    }
    onChange(e) {
        this.setState({ [e.target.name]: e.target.value })
    }
    onSubmit(e) {
        e.preventDefault();
        this.props.loginUser(this.state.username, this.state.password)

    }

    render() {

        if (this.props.auth ) {
            return <Redirect to="/" />
        }

        return (

            <form className="container form-login" onSubmit={this.onSubmit}>
                <div className="row">
                <div className="col s12 l8 offset-l2 m10 offset-m1">
                <h2>Login</h2>

                <div className="row">
                    <div className="input-field col s12">
                        <input id="email"  name="username" type="text" className="validate" value={this.state.username} onChange={this.onChange}/>
                        <label className="login-label-style" htmlFor="email">Email</label>
                    </div>
                </div>
                <div className="row">
                    <div className="input-field col s12 ">
                        <input id="password" type="password" className="validate" value={this.state.password} name="password" onChange={this.onChange} />
                        <label className="login-label-style" htmlFor="password">Password</label>
                    </div>
                </div>
                <div className="row">
                    <div className="input-field col s12 ">
                        {this.props.errors && this.props.errors.loginError?<h6 className="login-error">{this.props.errors.loginMessage}</h6>:null}
                    </div>
                </div>
                
                <button className="btn waves-effect waves-light black" type="submit" onClick={this.onSubmit} name="action">Submit
                    <i className="material-icons right">send</i>
                </button>
                </div>
                </div>
            </form>
        );
    }
}
function mapStateToProps(state) {
    return { auth:state.auth,errors:state.errors }
}

export default connect(mapStateToProps, { loginUser })(withRouter(Login));

