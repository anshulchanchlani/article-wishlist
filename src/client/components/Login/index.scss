.form-login{
    display:block;
    margin:auto
    label{
        color:rgb(49, 48, 48)
    }
}

.login-error{
    color:red;
}
.login-label-style{
    color:green!important
}