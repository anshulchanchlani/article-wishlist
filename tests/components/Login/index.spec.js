import React from 'react';
import {mockStore} from '../../setup/mockStore'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import {mount,shallow} from 'enzyme';
import toJson from 'enzyme-to-json'
import Login from '../../../src/client/components/Login'
import renderer from 'react-test-renderer'
const store = mockStore({wishlist:[],auth:'abc'})
describe('Login renders correctly',()=>{
    const component =  <Provider store={store}><BrowserRouter><Login auth='abc'/></BrowserRouter></Provider>
    const wrapper = shallow(component)
    it('Login renders correctly',()=>{
       
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})