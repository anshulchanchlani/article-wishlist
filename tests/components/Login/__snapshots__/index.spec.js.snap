// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly Login renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "auth": "abc",
        "wishlist": Array [],
      },
    }
  }
>
  <BrowserRouter>
    <Connect(withRouter(Login))
      auth="abc"
    />
  </BrowserRouter>
</ContextProvider>
`;
